c/src/nnstreamer-capi-pipeline.c

```diff
@@ -1426,12 +1428,14 @@ ml_pipeline_src_input_data (ml_pipeline_src_h h, ml_tensors_data_h data,
     ret = ML_ERROR_INVALID_PARAMETER;
     goto unlock_return;
   }
+  G_LOCK_UNLESS_NOLOCK (*_data);
 
   if (_data->num_tensors < 1 || _data->num_tensors > ML_TENSOR_SIZE_LIMIT) {
     ml_loge ("The tensor size is invalid. It should be 1 ~ %u; where it is %u",
         ML_TENSOR_SIZE_LIMIT, _data->num_tensors);
     ret = ML_ERROR_INVALID_PARAMETER;
-    goto destroy_data;
+    G_UNLOCK_UNLESS_NOLOCK (*_data);
```

Comment:
line 1446, 1456, 1468
requires unlock if policy == ML_PIPELINE_BUF_POLICY_DO_NOT_FREE

Reply:
Thanks. actually, they should not call destroy-data (another buf found!) if push-buffer is not called. Fix is coming.